### PR TITLE
Implement cross-thread postMessage

### PR DESCRIPTION
**Pull request by asajeffrey**
This PR implements cross-thread postMessage,

It builds on #15438 and #15478, only the last commit is part of this PR.

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] There are tests for these changes

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/15679)

### CODE REVIEW

**Comment by asajeffrey** [comment]
cc @jdm

**Comment by asajeffrey** [comment]
@emilio #15536 landed, this is now ready for review!

File: components/constellation/constellation.rs [hunk 1]
```diff
@@ -1795,6 +1799,21 @@ impl<Message, LTF, STF> Constellation<Message, LTF, STF>
         }
     }
 
+    fn handle_post_message_msg(&mut self, frame_id: FrameId, origin: Option<ImmutableOrigin>, data: Vec<u8>) {
+        let pipeline_id = match self.frames.get(&frame_id) {
+            None => return warn!("postMessage to closed frame {}.", frame_id),
```
Comment (emilio) [inline components/constellation/constellation.rs]: (As a side note, I seriously believe that we should convert all these to panics during testing).

File: tests/wpt/mozilla/tests/mozilla/cross-origin-objects/cross-origin-postMessage-child1.html [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<!DOCTYPE html>
+<html>
+<head>
```
Comment (emilio) [inline tests/wpt/mozilla/tests/mozilla/cross-origin-objects/cross-origin-postMessage-child1.html]: A lot of the tags here (and in the other tests) can go away and would make this a bit less noisy, but your choice :)

File: tests/wpt/mozilla/tests/mozilla/cross-origin-objects/cross-origin-postMessage-child1.html [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+<title> postMessage to dissimilar-origin iframe </title>
+</head>
+<body>
+<script>
+  window.addEventListener("message", function(e) {
+    window.location.href = e.data;
```
Comment (emilio) [inline tests/wpt/mozilla/tests/mozilla/cross-origin-objects/cross-origin-postMessage-child1.html]: heh. this is quite tricky. I guess it works :)

File: tests/wpt/mozilla/tests/mozilla/cross-origin-objects/cross-origin-postMessage.html [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+<body>
+<div id=log></div>
+<iframe id="child"></iframe>
+
+<script>
+  async_test(function(t) {
+    // child1 is a dissimilar-origin document
+    var childURL1 = new URL("cross-origin-postMessage-child1.html", document.location);
+    childURL1.hostname = "127.0.0.1";
+    // child2 is a same-orogin document
```
Comment (emilio) [inline tests/wpt/mozilla/tests/mozilla/cross-origin-objects/cross-origin-postMessage.html]: typo: `origin`.

**Review by emilio — APPROVED** [review]
r=me

File: components/constellation/constellation.rs (hunk 1 above)
Comment (asajeffrey) [inline components/constellation/constellation.rs]: That's an interesting idea. These warnings can happen during correct execution (e.g. post a message to an iframe's content window after it's been removed from the frame tree) but I don't think that should come up in wpt testing. File an issue?

File: tests/wpt/mozilla/tests/mozilla/cross-origin-objects/cross-origin-postMessage-child1.html (hunk 2 above)
Comment (asajeffrey) [inline tests/wpt/mozilla/tests/mozilla/cross-origin-objects/cross-origin-postMessage-child1.html]: I like my html old skool :)

File: tests/wpt/mozilla/tests/mozilla/cross-origin-objects/cross-origin-postMessage-child1.html (hunk 3 above)
Comment (asajeffrey) [inline tests/wpt/mozilla/tests/mozilla/cross-origin-objects/cross-origin-postMessage-child1.html]: I guess so :)

File: tests/wpt/mozilla/tests/mozilla/cross-origin-objects/cross-origin-postMessage.html (hunk 4 above)
Comment (asajeffrey) [inline tests/wpt/mozilla/tests/mozilla/cross-origin-objects/cross-origin-postMessage.html]: Oops!

**Review by asajeffrey — COMMENTED** [review]
Thanks for the quick review!

**Pull request merged**